src/SDK/Trace/Behavior/LoggerAwareTrait.php

```diff
@@ -23,20 +27,65 @@ public function setDefaultLogLevel(string $logLevel): void
         $this->defaultLogLevel = $logLevel;
     }
 
+    /**
+     * @return static
+     */
+    public function withLogger(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+
+        return $this;
+    }
+
     /**
      * @param string $message
      * @param array $context
      * @param string|null $level
      */
     protected function log(string $message, array $context = [], ?string $level = null): void
     {
+        //TODO basic info on which class created the message, cheaper than calculating from a stack trace
+        $context['source'] = __CLASS__;
```

Comment:
It's safer to use ``` $context['source'] = get_class($this); ``` here, since when for whatever reason a class is extended, ``` __CLASS__``` in a Trait will point to the class importing the trait.